src/Chapter1.hs

```diff
@@ -187,463 +184,460 @@ ghci> :q
+-- @
+--
+-- ♫ NOTE: __let-in__ is also an expression! You can't just define variables; you
+--   also need to return some expression that may use defined variables.
+--
+-- The syntax for defining multiple variables requires to care about indentation
+-- more, but there is nothing special in it as well:
+--
+-- @
+-- halfAndTwice :: Int -> (Int, Int)
+-- halfAndTwice n =
+--     let halfN = div n 2
+--         twiceN = n * 2
+--     in (halfN, twiceN)
+-- @
+--
+-- In addition to let-in (or sometimes even alternatively to let-in) you can use
+-- the __where__ construction to define local variables and functions.
+-- And, again, the example:
+--
+-- @
+-- pythagoras :: Double -> Double -> Double
+-- pythagoras a b = square a + square b
+--   where
+--     square :: Double -> Double
+--     square x = x ^ 2
+-- @
+--
+-- You can define multiple functions inside __where__!
+-- Just remember to keep proper indentation.
+
+-- |
+-- =⚔️= Task 9
+--
+-- Implement a function that returns the sum of the last two digits of a number.
+--
+-- >>> sumLast2 42
+-- 6
+-- >>> sumLast2 134
+-- 7
+-- >>> sumLast2 1
+-- 1
+--
+-- Try to introduce variables in this task (either with let-in or where) to avoid
+-- specifying complex expressions.
+sumLast2 :: Integral a => a -> a
+sumLast2 n = x + y
   where
-    square :: Double -> Double
-    square x = x ^ 2
-@
-
-You can define multiple functions inside __where__!
-Just remember to keep proper indentation.
--}
-
-{- |
-=⚔️= Task 9
-
-Implement a function that returns the sum of the last two digits of a number.
-
->>> sumLast2 42
-6
->>> sumLast2 134
-7
->>> sumLast2 1
-1
-
-Try to introduce variables in this task (either with let-in or where) to avoid
-specifying complex expressions.
--}
-
-sumLast2 n = error "sumLast2: Not implemented!"
-
-
-{- |
-=💣= Task 10*
-
-You did it! You've passed all the challenges in your first training!
-Congratulations!
-Now, are you ready for the boss at the end of this training???
-
-Implement a function that returns the first digit of a given number.
-
->>> firstDigit 230
-2
->>> firstDigit 5623
-5
-
-You need to use recursion in this task. Feel free to return to it later, if you
-aren't ready for this boss yet!
--}
-
-firstDigit n = error "firstDigit: Not implemented!"
-
-
+    an = abs n
+    x = an `mod` 10
+    y = (an `div` 10) `mod` 10
+
+-- |
+-- =💣= Task 10*
+--
+-- You did it! You've passed all the challenges in your first training!
+-- Congratulations!
+-- Now, are you ready for the boss at the end of this training???
+--
+-- Implement a function that returns the first digit of a given number.
+--
+-- >>> firstDigit 230
+-- 2
+-- >>> firstDigit 5623
+-- 5
+--
+-- You need to use recursion in this task. Feel free to return to it later, if you
+-- aren't ready for this boss yet!
+firstDigit :: Integral a => a -> a
+firstDigit n = go n 0
+  where
+    go 0 acc = abs acc
+    go x _ = go (ax `div` 10) (ax `mod` 10)
+      where
+        ax = abs x
```

Comment:
You can pass `abs n` to `go` in the first call of `go` on top-level. With this approach, you would only need to use `abs` once in the beginning and avoid calling it on each recursive call.

Reply:
:facepalm: That makes sense.  It is better to apply abs once than with every iteration of go.